crates/autopilot/src/run_loop.rs

```diff
@@ -680,9 +725,10 @@ impl RunLoop {
         let deadline = current.saturating_add(max_blocks_wait);
         tracing::debug!(%current, %deadline, %auction_id, "waiting for tag");
         loop {
-            if self.eth.current_block().borrow().number > deadline {
-                break;
-            }
+            let block = ethrpc::block_stream::next_block(self.eth.current_block()).await;
+            // Run maintenance to ensure the system processed the last available block so
+            // it's possible to find the tx in the DB in the next line.
+            self.run_maintenance().await;
```

Comment:
Can we somehow make sure we only run maintenance once per block hash (to avoid duplicate work)